test/powershell/Modules/Microsoft.PowerShell.Management/TimeZone.Tests.ps1

```diff
@@ -57,49 +56,53 @@ Describe "Get-Timezone test cases" -Tags "CI" {
     It "Call with ListAvailable switch returns a list containing TimeZoneInfo.Local" {
         $observedIdList = Get-TimeZone -ListAvailable | Select-Object -ExpandProperty Id
         $oneExpectedId = ([System.TimeZoneInfo]::Local).Id
-        $observedIdList -contains $oneExpectedId | Should Be $true
+        $observedIdList -eq $oneExpectedId | Should Be $oneExpectedId
     }
 
     It "Call with ListAvailable switch returns a list containing one returned by Get-TimeZone" {
         $observedIdList = Get-TimeZone -ListAvailable | Select-Object -ExpandProperty Id
         $oneExpectedId = (Get-TimeZone).Id
-        $observedIdList -contains $oneExpectedId | Should Be $true
+        $observedIdList -eq $oneExpectedId | Should Be $oneExpectedId
     }
 
     It "Call Get-TimeZone using ID param and single item" {
-        (Get-TimeZone -Id "Cape Verde Standard Time").Id -eq "Cape Verde Standard Time" | Should Be $true
+        $selectedTZ = $TimeZonesAvailable | Select-Object -Index ((Get-Random) % $TimeZonesAvailable.Count)
+        (Get-TimeZone -Id $selectedTZ.Id).Id | Should Be $selectedTZ.Id
```

Comment:
I think a reasonable test would be that if you get multiple timezones, check the first *n* (no need for random sample) and the break out of the test. No need to check them all. 
FWIW, this is still not deterministic because it relies on what is installed which may change across systems, and may not even satisfy the CI restart scenario if the config changes